applications/portal/backend/api/resources/AntibodyResource.py

```diff
@@ -0,0 +1,204 @@
+        super()
+        super().__init__()
+        self.antibody_identifiers = [
+            AntibodyIdentifier(
+                [self.fields['ab_id'], self.fields['ix'], self.fields['accession']],
+                lambda row: self.fields['ab_id'].column_name in row and (
+                        self.fields['ix'].column_name in row or self.fields['accession'].column_name in row),
+                lambda dataset: get_antibody_q1(dataset, self.fields['ab_id'],
+                                                [self.fields['ix'], self.fields['accession']]),
+                lambda dataset, negate, antibodies: filter_dataset_c1(dataset, negate, antibodies, self.fields['ab_id'],
+                                                                      self.fields['ix'], self.fields['accession'])
+            ),
+            AntibodyIdentifier(
+                [self.fields['vendor'], self.fields['catalog_num']],
+                lambda row: self.fields['vendor'].column_name in row and self.fields['catalog_num'].column_name in row,
+                lambda dataset: get_antibody_q2(dataset, self.fields['catalog_num'], self.fields['vendor']),
+                lambda dataset, negate, antibodies: filter_dataset_c2(dataset, negate, antibodies,
+                                                                      self.fields['catalog_num'],
+                                                                      self.fields['vendor'])
+            )
+        ]
+        self.request = request
+
+    class Meta:
+        model = Antibody
+        fields = (
+            'name', 'vendor', 'catalog_num', 'url', 'target', 'species', 'clonality', 'host', 'clone_id',
+            'product_isotype', 'product_conjugate', 'product_form', 'comments', 'defining_citation', 'subregion',
+            'modifications', 'gid', 'disc_date', 'commercial_type', 'uniprot', 'epitope', 'cat_alt', 'ab_id',
+            'accession', 'ix')
+        instance_loader_class = AntibodyInstanceLoaderClass
+
+    def get_antibody_identifier(self, row):
+        for antibody_identifier in self.antibody_identifiers:
+            if antibody_identifier.condition(row):
+                return antibody_identifier
+        return None
+
+    def get_or_init_instance(self, instance_loader, row):
+        """
+        Either fetches an already existing instance or initializes a new one.
+        """
+        create_duplicate = self.request.session.get(FOR_EXTANT_KEY, UPDATE_KEY) == DUPLICATE_KEY
+
+        if not self._meta.force_init_instance:
+            instance = self.get_instance(instance_loader, row)
+            if instance:
+                if not create_duplicate:
+                    return (instance, False)
+                row[self.fields['accession'].column_name] = instance.ab_id
+                # todo: confirm if the following is suppose to happen
+                if self.fields['ab_id'].column_name in row:
+                    del row[self.fields['ab_id'].column_name]
```

Comment:
@filippomc to confirm if when the create duplicate option is selected we want the ab_id of the duplicate to change (I think it only makes sense that we do but it's not clear from the card)